pkg/operator/ceph/cluster/osd/deviceSet.go

```diff
@@ -308,3 +336,67 @@ func createValidImageVersionLabel(image string) string {
 	}
 	return cephImageVersion
 }
+
+func waitForPvcToExpandWithTimeout(ctx context.Context, client client.Client, pvcResizeMap map[string]pvcResize, namespace string, pvcResizeWaitTimeOut time.Duration) {
+	currentTime := time.Now().UTC()
+	if pvcResizeWaitTimeOut == 0 {
+		pvcResizeWaitTimeOut = defaultWaitTimeoutForHealthyOSD
+	}
+	timeout := currentTime.Add(pvcResizeWaitTimeOut)
+	for {
+		if time.Now().UTC().After(timeout) {
+			logger.Errorf("timed out waiting for PVC size to be updated, osds may need to be restarted manually after resize")
+			return
+		}
+
+		allPvcResize := checkAllPvcResize(ctx, client, namespace, pvcResizeMap)
+		if allPvcResize {
+			logger.Info("all osd pvcs resized")
+			return
+		}
+
+		// wait for 5 seconds before checking again
+		time.Sleep(5 * time.Second)
+	}
+}
+
+func checkAllPvcResize(ctx context.Context, client client.Client, namespace string, pvcResizeMap map[string]pvcResize) bool {
+	numberOfPvcResize := 0
+
+	for pvcName, pvc := range pvcResizeMap {
+		if pvc.resized {
+			numberOfPvcResize++
+			continue
+		}
+
+		// get the latest PVC
+		currentPVC := &v1.PersistentVolumeClaim{}
+		err := client.Get(ctx, types.NamespacedName{Name: pvcName, Namespace: namespace}, currentPVC)
+		if err != nil {
+			logger.Errorf("failed to get PVC %q. %v", currentPVC.Name, err)
+			continue
+		}
+
+		// check if the size is updated
+		pvcSize := currentPVC.Spec.Resources.Requests[v1.ResourceStorage]
+		if pvcSize.Cmp(pvc.desiredSize) != 0 {
+			continue
+		}
+
+		if currentPVC.Status.Capacity != nil {
+			if currentPVC.Status.Capacity[v1.ResourceStorage] == pvcSize {
+				logger.Infof("PVC %q size is updated to %s", currentPVC.Name, pvcSize.String())
+				numberOfPvcResize++
+				pvcResizeMap[pvcName] = pvcResize{
+					desiredSize: pvc.desiredSize,
+					actualSize:  pvc.actualSize,
+					resized:     true,
+				}
+			} else {
+				logger.Infof("PVC %q size is not updated to %v", currentPVC.Name, pvcSize.String())
+			}
+		}
+	}
+
```

Comment:
One summary log message per loop should be good for info level. How about logging how many OSDs have been resized and how many remain?